.github/workflows/ci.yml

```diff
@@ -0,0 +1,48 @@
+on: pull_request
+
+name: CI
+jobs:
+  build_and_test:
+    name: Build and test on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+
+    steps:
+      - uses: actions/checkout@master
+
+      - name: Install nightly
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          override: true
+
+      - name: check
+        uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --all --benches --bins --examples --tests
+
+      - name: tests
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --all --doc
```

Comment:
Run doc-tests only?

Reply:
But tests under the `tests` dir are very important. Checks for some api like shutdown.

Reply:
I think we can just use `--all`

Reply:
ah yes better.